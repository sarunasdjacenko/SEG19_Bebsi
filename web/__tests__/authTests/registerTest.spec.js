import { shallowMount } from "@vue/test-utils";
import Vue from "vue";
import VueRouter from "vue-router";
// @ts-ignore
import Component from "../../src/components/auth/Register.vue";
Vue.use(VueRouter);

describe("Component", () => {
    const wrapper = shallowMount(Component);
    test("is a Vue instance", () => {
      expect(wrapper.isVueInstance()).toBeTruthy();
    }),
      //snapshot test
      test("renders correctly", () => {
        expect(wrapper.element).toMatchSnapshot();
      }),
      test("has the correct buttons rendered", () => {
        const registerBtn = wrapper.find("#registerBtn");
        const checkBox = wrapper.find("#check");
  
        expect(registerBtn).toBeDefined();
        expect(checkBox).toBeDefined();
      }),
      test("form submits successfully", () => {
        wrapper.setMethods({ register: jest.fn() });
        wrapper.find("#registerBtn").trigger("submit");
  
        expect(wrapper.vm.register).toHaveBeenCalled();
      }),
      test("has the right amount of inputs", () => {
          expect(wrapper.findAll("input")).toHaveLength(5);
      }),
      test("all inputs have required attribute", () => {
        var inputArray = wrapper.findAll("input");
        // get if there is an input without the required property
        inputArray = inputArray.filter(input => !input.attributes("required"));
        expect(inputArray).toHaveLength(1);
      });
  });