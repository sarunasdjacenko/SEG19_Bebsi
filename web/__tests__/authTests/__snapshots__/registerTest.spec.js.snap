// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<div>
  <div
    class="container"
    style="padding-top: 10em;"
  >
    <div
      class="row"
    >
      <div
        class="col s12 m8 offset-m2"
      >
        <div
          class="login card-panel white blue-text center"
        >
          <h3>
            Register
          </h3>
           
          <form>
            <div
              class="input-field"
            >
              <i
                class="material-icons prefix"
              >
                email
              </i>
               
              <input
                id="email"
                name="email"
                placeholder="Email"
                required="required"
                type="text"
              />
            </div>
             
            <div
              class="input-field"
            >
              <i
                class="material-icons prefix"
              >
                lock
              </i>
               
              <input
                id="password"
                name="password"
                placeholder="Password"
                required="required"
                type="password"
              />
            </div>
             
            <div
              class="input-field"
            >
              <i
                class="material-icons prefix"
              >
                face
              </i>
               
              <input
                id="first"
                name="first"
                placeholder="Name"
                required="required"
                type="text"
              />
            </div>
             
            <div
              class="input-field"
            >
              <i
                class="material-icons prefix"
              >
                assignment
              </i>
               
              <input
                id="department"
                name="department"
                placeholder="Department"
                required="required"
                type="text"
              />
            </div>
             
            <div
              class="input-field"
            >
              <p
                style="margin-right: 100%;"
              >
                <i
                  class="material-icons prefix blue-text"
                  style="margin-left: 8px;"
                >
                  assignment_ind
                </i>
                 
                <label
                  style="border-bottom: 1px solid #9e9e9e;"
                >
                  <input
                    class="filled-in"
                    id="check"
                    style="margin-top: 10px;"
                    type="checkbox"
                  />
                   
                  <span
                    class="blue-text"
                    style="margin-top: 10px;"
                  >
                    Admin
                  </span>
                </label>
              </p>
            </div>
             
            <button
              class="btn btn-large blue lighten-2 black-text"
              id="registerBtn"
              style="margin-top: 70px;"
              type="submit"
            >
              Register
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
